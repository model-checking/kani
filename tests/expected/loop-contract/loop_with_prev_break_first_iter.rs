// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT

// kani-flags: -Z loop-contracts

//! Check if loop contracts is correctly applied.

#![feature(stmt_expr_attributes)]
#![feature(proc_macro_hygiene)]

#[kani::proof]
pub fn loop_with_prev() {
    let mut i = 100;
    #[kani::loop_invariant((i >= 2) && (i <= 100) && (prev(i) == i + 2))]
    while i > 2 {
        if i == 100 {
            break;
        }
        i = i - 2;
    }
    assert!(i == 2);
}
