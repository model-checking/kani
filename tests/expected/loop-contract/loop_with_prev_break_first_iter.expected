Failed Checks: assertion failed: (i >= 2) && (i <= 100) && (__kani_prev_var_

VERIFICATION:- FAILED
